server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -1,167 +1,117 @@
+    }
+  }
+
+  implicit class AutoSource(searchDefinition: SearchDefinition) {
+    def format(implicit optFmt: Option[AddressScheme]) = {
+      optFmt map {
+        case _: PostcodeAddressFile => searchDefinition sourceExclude HybridIndex.Fields.lpi
+        case _: BritishStandard7666 => searchDefinition sourceExclude HybridIndex.Fields.paf
+      } getOrElse searchDefinition
+    }
+  }
+}
+
 @ImplementedBy(classOf[AddressIndexRepository])
-trait ElasticsearchRepository {
+trait ElasticSearchRepository {
+
   /**
     * An ElasticClient.
     */
-  def client() : ElasticClient
+  def client: ElasticClient
 
-  /**
-    * Query the PAF addres index by UPRN.
-    *
-    * @param uprn the identificator of the address
-    * @return Fucture containing a PAF address or `None` if not in the index
-    */
-  def queryPafUprn(uprn : String) : Future[Option[PostcodeAddressFileAddress]]
+  def logger: Logger
 
   /**
-    * Query the NAG addres index by UPRN.
     *
-    * @param uprn the identificator of the address
-    * @return Fucture containing a NAG address or `None` if not in the index
+    * @param tokens
+    * @param p
+    * @param fmt
+    * @return
     */
-  def queryNagUprn(uprn : String) : Future[Option[NationalAddressGazetteerAddress]]
+  def queryAddress(tokens: Seq[CrfTokenResult])
+    (implicit p: Pagination, fmt: Option[AddressScheme]): Future[RichSearchResponse] = {
+    logExecute("Address")(queryAddressSearchDefinition(tokens))
+  }
 
   /**
-    * Query the address index for PAF addresses.
-    * Currently the query must be for building number and postcode.
     *
-    * @param tokens address tokens
-    * @return Future with found PAF addresses and the maximum score
+    * @param uprn
+    * @param fmt
+    * @return
     */
-  def queryPafAddresses(start: Int, limit: Int, tokens: Seq[CrfTokenResult]) : Future[PostcodeAddressFileAddresses]
+  def queryUprn(uprn: String)(implicit fmt: Option[AddressScheme]): Future[RichSearchResponse] = {
+    logExecute("UPRN")(queryUprnSearchDefinition(uprn))
+  }
+
+  def queryUprnSearchDefinition(uprn: String)(implicit fmt: Option[AddressScheme]): SearchDefinition
+
+  def queryAddressSearchDefinition(tokens: Seq[CrfTokenResult])
+    (implicit p: Pagination, fmt: Option[AddressScheme]): SearchDefinition
 
   /**
-    * Query the address index for NAG addresses.
-    * Currently the query must be for building number and postcode.
+    * Helper which logs an ElasticSearch Query before executing it.
     *
-    * @param tokens address tokens
-    * @return Future with found PAF addresses and the maximum score
+    * @param info
+    * @param t
+    * @param executable
+    * @tparam T
+    * @tparam R
+    * @tparam Q
+    * @return
     */
-  def queryNagAddresses(start: Int, limit: Int, tokens: Seq[CrfTokenResult]) : Future[NationalAddressGazetteerAddresses]
+  def logExecute[T, R, Q](info: String)(t: T)(implicit executable: Executable[T, R, Q]): Future[Q] = {
```

Comment:
This is a wrapper on the Elastic4s Client, nothing fancy going on here just a copy of the Elastic4s's Client.execute method 

The idea is that you use this method instead, and we log the elastic4s `json` dynamically

also the DSL looks nice :)